### ISSUE TITLE
Dropout in last layer

### ISSUE THREAD

**Issue by foxtrotmike** [weihua916/powerful-gnns#16]
Thank you for your work. 
You have used dropout prior to computing the output from each layer. What is the role of this dropout?
See: https://github.com/weihua916/powerful-gnns/blob/f2626e7ed9c81d40e315f657c9c5c1611e071cc9/models/graphcnn.py#L225

**Comment by weihua916** [comment]
It's just a regularization.

**Comment by foxtrotmike** [comment]
Thank you for your response - Tome, it makes sense to regularize using dropout at layers before the last one. Dropout at the prediction output forces prediction scores of an example to become zero and also scales up the prediction scores of other examples in the batch. Please see: "Furthermore, the outputs are scaled by a factor of \frac{1}{1-p}" in https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html
It would be great if you could help me understand this a bit better. Thanks, again.
